bundles/test/org.eclipse.smarthome.magic/src/main/java/org/eclipse/smarthome/magic/binding/internal/MagicServiceImpl.java

```diff
@@ -49,4 +56,14 @@
         return null;
     }
 
+    @Activate
+    public void activate(Map<String, Object> properties) {
+        modified(properties);
+    }
+
+    @Modified
+    public void modified(Map<String, Object> properties) {
+        MagicServiceConfig config = new Configuration(properties).as(MagicServiceConfig.class);
```

Comment:
Since you introduced the new ConfigUtil#as method you can directly use it here.

Reply:
That would be the only place of all services where it would be different. I'd like to wait for the discussion and outcome of #5300, if that is ok for you.

Reply:
agreed.